contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsFetcherAggStats.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    public synchronized void removeStoreCopyStats(String storeName) {
+        copyStatsMap.remove(storeName);
+    }
+
+    public synchronized HdfsCopyStats getStoreCopyStatus(String storeName) {
+        return copyStatsMap.get(storeName);
+    }
+
+    @JmxGetter(name = "totalBytesFetched", description = "The total bytes transferred from HDFS so far.")
+    public synchronized long getTotalBytesFetched() {
+        return totalBytesFetched;
+    }
+
+    @JmxGetter(name = "totalFetchRetries", description = "The total fetch retry number so far.")
+    public synchronized long getTotalFetchRetries() {
+        return totalFetchRetries;
+    }
+
+    @JmxGetter(name = "totalCheckSumFailures", description = "The total data file checksum failures happened so far.")
+    public synchronized long getTotalCheckSumFailures() {
+        return totalCheckSumFailures;
+    }
+
+    @JmxGetter(name = "totalAuthenticationFailures", description = "The total authentication failures happened so far.")
+    public synchronized long getTotalAuthenticationFailures() {
+        return totalAuthenticationFailures;
+    }
+
+    @JmxGetter(name = "totalFileNotFoundFailures", description = "The total file-not-found failures happened so far.")
+    public synchronized long getTotalFileNotFoundFailures() {
+        return totalFileNotFoundFailures;
+    }
+
+    @JmxGetter(name = "totalFileReadFailures", description = "The total HDFS file read failures happened so far.")
+    public synchronized long getTotalFileReadFailures() {
+        return totalFileReadFailures;
+    }
+
+    @JmxGetter(name = "totalQuotaExceedFailures", description = "The total quota exceed failures happened so far.")
+    public synchronized long getTotalQuotaExceedFailures() {
+        return totalQuotaExceedFailures;
+    }
+
+    @JmxGetter(name = "totalUnauthorizedStoreFailures", description = "The total unauthorized store push failures happened so far.")
+    public synchronized long getTotalUnauthorizedStoreFailures() {
+        return  totalUnauthorizedStoreFailures;
+    }
+
+    @JmxGetter(name = "parallelFetches", description = "The total number of active fetches right now.")
+    public synchronized int getParallelFetches() {
+        return parallelFetches;
+    }
+
+    @JmxGetter(name = "totalFetches", description = "The total HDFS fetch number so far.")
+    public synchronized long getTotalFetches() {
+        return totalFetches;
+    }
+
+    @JmxGetter(name = "totalIncompleteFetches", description = "The total incomplete fetch number so far.")
+    public synchronized long getTotalIncompleteFetches() {
+        return totalIncompleteFetches;
+    }
+
+    @JmxGetter(name = "totalDataFetchRate", description = "The total data fetch rate right now.")
+    public synchronized double getTotalDataFetchRate() {
+        double fetchRate = 0;
+        for (HdfsCopyStats copyStats : copyStatsMap.values()) {
+            fetchRate += copyStats.getBytesTransferredPerSecond();
```

Comment:
The fetchRate is not often updated and might be calculated at different intervals. EventThrottler has a rate sensor, we should output that for totalFetchRate.